src/styles/main.scss

```diff
@@ -1,3 +1,54 @@
+$day-size: 100px;
+
 body {
   margin: 0;
 }
+
+.calendar {
+  width: $day-size * 7 + 20px;
+  padding: 10px;
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+
+  &__day {
+    box-sizing: border-box;
+    height: $day-size;
+    width: $day-size;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+
+    background-color: #eee;
+    border: 1px solid black;
+    transition-duration: 0.5s;
+
+    &:hover {
+      cursor: pointer;
+      background: pink;
+      transform: translate(0, -20px);
+    }
+  }
+}
+
+@for $i from 1 through 31 {
+  .calendar__day:nth-child(#{$i})::before {
+    content: "#{$i}";
+    font-family: Arial, Helvetica, sans-serif;
+    font-size: 30px;
+  }
+}
```

Comment:
Checklist:
7. [SASS] - Don't use SASS loops for styles that stay the same for all elements of the group, e.g. display or position.

Also, you can use loops inside selectors and use the advantages of preprocessors such as nesting.